packages/ag-charts-types/src/chart/themeOptions.ts

```diff
@@ -78,9 +78,31 @@ export interface AgChartThemePalette {
     neutral?: AgPaletteColors;
 }
 
+export interface AgChartThemeParams {
+    /** TODO */
+    fontFamily?: string;
+    /** TODO */
+    fontSize?: number;
+}
+
+type ExtendLiteralLeaves<T, V> = {
+    [P in keyof T]: T[P] extends (infer U)[]
+        ? ExtendLiteralLeaves<U, V>[]
+        : T[P] extends string | symbol | number | undefined
+          ? T[P] | V
+          : ExtendLiteralLeaves<T[P], V>;
+};
+
+type ThemeParam = keyof AgChartThemeParams;
+type ThemeParamsRef = { ref: ThemeParam };
+
+export type WithThemeParams<T> = ExtendLiteralLeaves<T, ThemeParamsRef>;
```

Comment:
For now, this type is only used internally. It wraps the series modules and annotations theme definitions, so we can use `as const` instead of `as unknown as string`.